### ISSUE TITLE
False negative on Slim's text interpolation syntax?

### ISSUE THREAD

**Issue by neko314** [r7kamura/slimcop#8]
`Layout/SpaceInsideStringInterpolation` is enable by default on RuboCop. (See: https://docs.rubocop.org/rubocop/cops_layout.html#layoutspaceinsidestringinterpolation).
I expect slimcop to find offending codes but it doesn't. RuboCop finds offense in anything but views (like models) so I think there is something wrong with slimcop or templatecop but I'm not sure🤔

## Step to reproduce

```yml
# rubocop.yml
# no matter if I specify configuration or not. just example below
Layout/SpaceInsideStringInterpolation
  enable: true
  EnforcedStyle: no_space
```

```
# app/views/sample.html.slim 
- name = "name"
= "#{ name }"
```

and run checking

```
slimcop "app/views/sample.html.slim"
```

## Expected result

```
app/views/sample.html.slim:n:n C: Layout/SpaceInsideStringInterpolation: Space inside string interpolation detected.
```

## Actual result

```
1 file inspected, no offenses detected
```

**Comment by r7kamura** [comment]
> ```
> # app/views/sample.html.slim 
> = #{ name }
> ```

@neko314 Are you sure it is really `= #{ name }`, not `= "#{ name }"` mistake?

I think that example should cause SyntaxError on the rendering phase like this... or am I misunderstanding something?

```console
$ echo '= #{ 1 }' | slimrb --stdin 
SyntaxError: (__TEMPLATE__):4: syntax error, unexpected `end', expecting ')'
end;end;
^~~
  Use --trace for backtrace.
$ echo '= "#{ 1 }"' | slimrb --stdin 
1
```

**Comment by neko314** [comment]
@r7kamura 
Oh, sorry for my mistakes.  I wrote code example based on actual code I have locally, and neglected to check sysntax.

I updated the description.

**Comment by r7kamura** [comment]
@neko314 I prepared a Slim template with the same content and ran Slimcop, but it did not cause any problems. Can you tell me exactly how to reproduce this in any way?

```console
$ cat example.slim
- name = "name"
= "#{ name }"
$ slimcop example.slim
Inspecting 1 file
C

Offenses:

example.slim:1:10: C: [Correctable] Style/StringLiterals: Prefer single-quoted strings when you don't need string interpolation or special symbols.
- name = "name"
         ^^^^^^
example.slim:2:6: C: [Correctable] Layout/SpaceInsideStringInterpolation: Space inside string interpolation detected.
= "#{ name }"
     ^
example.slim:2:11: C: Layout/SpaceInsideStringInterpolation: Space inside string interpolation detected.
= "#{ name }"
          ^
example.slim:2:3: C: [Correctable] Style/RedundantInterpolation: Prefer to_s over string interpolation.
= "#{ name }"
  ^^^^^^^^^^^

1 file inspected, 4 offenses detected, 3 offenses auto-correctable
```

**Comment by r7kamura** [comment]
I found that the reported example is not good enough, but with `p #{ a}`, I could reproduce this problem.

```console
$ cat example.slim
p #{ a}
$ slimcop example.slim
Inspecting 1 file
.

1 file inspected, no offenses detected
```

**Comment by r7kamura** [comment]
This is debug information from the parser, posibly helpful:

```console
$ slimi parse <<< "p #{ a}"
[:multi,
 [:html,
  :tag,
  "p",
  [:html, :attrs],
  [:slimi, :text, :inline, [:multi, [:slimi, :interpolate, 2, 7, "\#{ a}"]]]],
 [:newline]]
```

**Comment by r7kamura** [comment]
I finally understood. This `#{ ... }` is a special syntax on Slim, not Ruby's interpolation in String, and therefore slimcop cannot detect this kind of pattern by RuboCop.

**Comment by r7kamura** [comment]
Slimcop runs RuboCop on the embedded Ruby code, and is not a tool to detect whitespace, etc. in the Slim syntax, so it cannot detect this type of problem.

As another perspective, one might ask if it is possible to remove the leading whitespace in the embedded Ruby code of "    a". This can be detected by `Layout/InitialIndentation` Cop for normal Ruby code, but for Ruby code embedded in Slim, there are many cases where leading whitespace is placed at the beginning due to Slim coding style, and currently there is no mechanism to distinguish it from them. This is why we have no choice but to disable this Cop.

**Comment by r7kamura** [comment]
The problem can be detected by enabling Layout/InitialIndentation as follows:

```console
$ cat example.slim 
p #{ a}
$ cat .slimcop.yml 
Layout/InitialIndentation:
  Enabled: true
$ slimcop
Inspecting 1 file
C

Offenses:

example.slim:1:6: C: [Correctable] Layout/InitialIndentation: Indentation of first line in file detected.
p #{ a}
     ^

1 file inspected, 1 offense detected, 1 offense auto-correctable
```

I think I disabled it for some reason (maybe copied from erbcop?), but I should stop disabling Layout/InitialIndentation.

**Comment by r7kamura** [comment]
Enabled Layout/InitialIndentation at https://github.com/r7kamura/slimcop/pull/9, and released v0.17.0 👍

**Comment by neko314** [comment]
Thank you so much!
